### DISCUSSION TITLE
Can babel makes export default to fit commonjs or AMD module?

### DISCUSSION THREAD

**Discussion by TheColdPot** [babel/babel#12376] [Q&A]
I am using webpack+babel to package my project. Is there a plug-in that can automatically insert and export the automatic recognition module? Or, is there a new version of the plug-in that can add `module.exports` again (`babel-plugin-add-module-exports`?)

**Comment by nicolo-ribaudo** [comment]
Not currently, I opened an issue about CJS at https://github.com/babel/babel/issues/12363

**Reply by TheColdPot** [reply]
OK. But what if I write code my own like
```js
if (typeof define === 'object' && defined.amd) define( /* PARAMS */ );
if (typeof module === 'object' && module.exports) module.exports = /* OBJ */;
globalThis./* OBJ */ = /* OBJ */
```
